src/mountscript.sh

```diff
@@ -160,6 +163,31 @@ check_nconnect()
                     vvecho "Azure nconnect not enabled, enabling!"
                     echo Y > /sys/module/sunrpc/parameters/enable_azure_nconnect
                 fi
+
+                # Check if AZNFS_MAX_NCONNECT is defined, numeric, and within the allowed range.
+                if [[ "$AZNFS_MAX_NCONNECT" =~ ^[0-9]+$ && "$AZNFS_MAX_NCONNECT" != "16" ]]; then
+
+                    if [[ "$AZNFS_MAX_NCONNECT" -lt 1 || "$AZNFS_MAX_NCONNECT" -gt 16 ]]; then
+                        eecho "[ERROR] Incorrect value $AZNFS_MAX_NCONNECT for the environment variable AZNFS_MAX_NCONNECT"
+                        exit 1
+                    fi
+
+                    # We reach here only if nconnect matches the regex 'matchstr="\<nconnect\>=([0-9]+)"'
+                    if [[ "$value" -lt 1 || "$value" -gt 16 ]]; then
+                        eecho "[ERROR] nconnect value must be provided between 1 and $AZNFS_MAX_NCONNECT"
+                        exit 1
+                    fi
+
+                    # Re-calculate the maximum accounts mountable from a single tenant.
+                    MAX_ACCOUNTS_MOUNTABLE_FROM_SINGLE_TENANT=$((320 / AZNFS_MAX_NCONNECT))
+
+                    # Check if the current nconnect value in use is suboptimal.
+                    if [[ "$value" -gt "$AZNFS_MAX_NCONNECT" ]]; then
+                        pecho "Suboptimal nconnect value $value, forcing nconnect=$AZNFS_MAX_NCONNECT!"
```

Comment:
this log should say limiting nconnect to such and such as advised by AZNFS_MAX_NCONNECT